labs/database/vectorize/python_vectorizer.py

```diff
@@ -0,0 +1,137 @@
+from types import SimpleNamespace
+
+from langchain_community.document_loaders import TextLoader
+from langchain_core.documents import Document
+from litellm import embedding
+import openai
+import os
+import pathspec
+
+import logging
+
+from labs.config import settings
+from labs.database.embeddings import reembed_code
+
+from labs.parsers.python import get_lines_code, parse_python_file
+from labs.database.vectorize import Vectorizer
+
+logger = logging.getLogger(__name__)
+
+openai.api_key = settings.OPENAI_API_KEY
+
+
+class PythonVectorizer(Vectorizer):
+    def prepare_doc_content(self, metadata, code_snippet):
+        metadata = SimpleNamespace(**metadata)
```

Comment:
Python newbie: Where does this "SimpleNamespace" comes from?

Reply:
SimpleNamespace comes from "types" and allows you to use a dictionary as an object with properties like:

`ns = SimpleNamespace(**dict(prop1="prop1val", prop2="prop2val")
  val1 = ns.prop1
  val2 = ns.prop2
`